### ISSUE TITLE
V6: Error in code after an async validation results in swallowed Errors (maybe bluebird specific)

### ISSUE THREAD

**Issue by q42jaap** [erikras/redux-form#1201]
https://github.com/erikras/redux-form/blob/v6/src/reduxForm.js#L226

Let's say I make a typo in my onSubmit() function, causing a TypeError or something similar.
The resulting error causes a rejected promise (this might depend on promise library, bluebird wraps all continuations with tryCatcher).
The listenToSubmit function handles the submit promise (which is a result of having async validation).

The second argument to `.then()` "handles" the error.

I would like to suggest to have a different approach (or something similar which allows binding to `this`):

```
return promise.then(this.submitCompleted, (err) => {
  this.submitCompleted();
  return Promise.reject(err);
})
```

This way the Promise library taking the `then` can handle the rejection as needed.

This works well with bluebird, and I expect it to also work with native promises.

**Comment by erikras** [comment]
That seems reasonable.

**Comment by erikras** [comment]
Addressed in #1203. Will be in next release.
